src/main/java/com/sk89q/craftbook/util/ItemUtil.java

```diff
@@ -940,10 +941,64 @@ public static short getMaxDurability(Material type) {
                 return type.getMaxDurability();
         }
     }
+    
+    public static boolean isArmor(Material type){
+        switch(type) {
+            case LEATHER_HELMET:
+            case LEATHER_CHESTPLATE:
+            case LEATHER_LEGGINGS:
+            case LEATHER_BOOTS:
+            case IRON_HELMET:
+            case IRON_CHESTPLATE:
+            case IRON_LEGGINGS:
+            case IRON_BOOTS:
+            case GOLDEN_HELMET:
+            case GOLDEN_CHESTPLATE:
+            case GOLDEN_LEGGINGS:
+            case GOLDEN_BOOTS:
+            case DIAMOND_HELMET:
+            case DIAMOND_CHESTPLATE:
+            case DIAMOND_LEGGINGS:
+            case DIAMOND_BOOTS:
+            case CHAINMAIL_HELMET:
+            case CHAINMAIL_CHESTPLATE:
+            case CHAINMAIL_LEGGINGS:
+            case CHAINMAIL_BOOTS:
+            case TURTLE_HELMET:
+                return true;
+            default:
+                return false;
+        }
+    }
+    
+    public static int getUnbreakingEnchantLevel(ItemStack stack) {
+        Map<Enchantment, Integer> enchants = stack.getEnchantments();
+        return enchants.getOrDefault(Enchantment.DURABILITY, 0);
+    }
+    
+    public static boolean doesUnbreakingNotPreventLoss(int level, boolean isArmor){
+        // general case is (100 / (level + 1))% chance
+        int chance = (int)(100d / (level + 1));
```

Comment:
Is there a reason you convert this back into integers, rather than just using a double for the entire calculation? This somewhat will affect the probabilities